### PR TITLE
BUG: Fix frame column validation in Python 2.7.5 (#1771)

### PR DESCRIPTION
**Pull request by tibkiss**

### CODE REVIEW

File: zipline/data/us_equity_pricing.py [hunk 1]
```diff
@@ -915,7 +915,8 @@ def _write(self, tablename, expected_dtypes, frame):
                 np.array([], dtype=list(expected_dtypes.items())),
             )
         else:
-            if frozenset(frame.columns) != viewkeys(expected_dtypes):
+            if frozenset(frame.columns) != \
+               frozenset(viewkeys(expected_dtypes)):
```
Comment (ssanderson) [inline zipline/data/us_equity_pricing.py]: Is there still a point in calling `viewkeys` here if we're just going to pass the input to `frozenset` anyway?

**Comment by ssanderson** [comment]
@tibkiss this seems reasonable to me.  I assume this is working around https://bugs.python.org/issue8404 ?  Just had one small perf-related question.

**Comment by richafrank** [comment]
Looks like a fix for https://github.com/quantopian/zipline/issues/1641

File: zipline/data/us_equity_pricing.py (hunk 1 above)
Comment (tibkiss) [reply zipline/data/us_equity_pricing.py]: good point, updated the PR accordingly

**Comment by tibkiss** [comment]
@richafrank / @ssanderson : Thanks for the quick review & sorry for the late response. 

@ssanderson : Even though your linked issue looks really similar I don't think it is exactly the same. In issue8404 the affectedVersion is a Python RC and the problem I'm workarounding here is present on python 2.7.5. 

Nonetheless this fix solves #1771 & #1641 (which are essentially the same).

**Comment by richafrank** [comment]
CI tests pass, and this looks safe to me. Merging.

**Comment by richafrank** [comment]
Thanks @tibkiss !

**Pull request merged**